src/implementation/Client/GRPCClient/GRPCClient.ts

```diff
@@ -29,17 +31,34 @@ export default class GRPCClient implements IClient {
   private readonly clientCredentials: grpc.ChannelCredentials;
   private readonly logger: Logger;
   private readonly grpcClientOptions: Partial<grpc.ClientOptions>;
+  private daprEndpoint: GrpcEndpoint;
+
+  constructor(options: Partial<DaprClientOptions>) {
+    this.daprEndpoint = this.generateEndpoint(options);
+
+    this.options = {
+      daprHost: options?.daprHost || this.daprEndpoint.hostname,
```

Comment:
Do we need to check for `options?.daprHost`, can `generateEndpoint` handle this since it already has `options`?